hip-clang.docker

```diff
@@ -86,3 +88,17 @@ RUN pip install -r /doc-requirements.txt
 # install last released miopentensile in default, install latest commits when MIOTENSILE_VER="latest"
 RUN if [ "$MIOTENSILE_VER" = "latest" ] ; then cget -p $PREFIX install ROCmSoftwarePlatform/MIOpenTensile@7568654c938d42e9a91c6b18fb382f5b978d12fd; else cget -p $PREFIX install ROCmSoftwarePlatform/MIOpenTensile@5fe0bf4a8dc59f3ab62df929297280915372ce16; fi
 
+RUN cd ~ && \
+    export MLIR_COMMIT=06a6d3fc1af1fdfbe706dbfcd588fb3744526124 && \
+    wget https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/archive/$MLIR_COMMIT.tar.gz && \ 
+    tar -xvzf $MLIR_COMMIT.tar.gz && \
+    rm -rf $MLIR_COMMIT.tar.gz && \
+    cd llvm-project-mlir-$MLIR_COMMIT && mkdir -p build && cd build && \
+    $PREFIX/bin/cmake -G "Unix Makefiles" ../llvm \
+      -DLLVM_ENABLE_PROJECTS="mlir;lld" \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBUILD_SHARED_LIBS=OFF \
+      -DLLVM_BUILD_LLVM_DYLIB=OFF \
+      -DLLVM_ENABLE_TERMINFO=OFF && \
+    make -j libMLIRMIOpen && \
```

Comment:
Can this be installed with cget instead with:

```
cget -p $PREFIX install ROCmSoftwarePlatform/llvm-project-mlir@06a6d3fc1af1fdfbe706dbfcd588fb3744526124 -X subdir -DCMAKE_DIR=llvm -DLLVM_ENABLE_PROJECTS="mlir;lld" -DBUILD_SHARED_LIBS=OFF -DLLVM_BUILD_LLVM_DYLIB=OFF -DLLVM_ENABLE_TERMINFO=OFF
```

When we move to use this in our releases, we will list this in the requirements.txt file so it will automatically get build during our package building. So it would be good to make sure it works with cget already.

It would be even nicer if llvm-project-mlir could provide a top-level cmake file that sets all these flags and then does `add_subdirectory(llvm)`, so we could just install simply as `cget -p $PREFIX install ROCmSoftwarePlatform/llvm-project-mlir@06a6d3fc1af1fdfbe706dbfcd588fb3744526124`

Reply:
It'd be nice if this can work with cget but I just experimented your suggestion real quick and it turned out build did not finish. Considering for the short term this MLIR feature remains experimental, I don't see it necessarily go in this review. Are you okay with me doing the following:
 - Skip this comment for this PR
 - File an issue on MLIR side tracking the cget build, referencing your comments here
 - When this issue get fixed in MLIR side, file another PR to bump the commit && use cget here

Reply:
> Are you okay with me doing the following

Yes that is acceptable.

Reply:
MLIR https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/56 updated. Marking this conversation as resolved.